Sources/TuistKit/Services/ShareService.swift

```diff
@@ -358,15 +358,19 @@ struct ShareService {
                 }
                 .uniqued()
 
+            let appBundles = try await appPaths.concurrentMap {
+                try await appBundleLoader.load($0)
+            }
+
             if appPaths.isEmpty {
                 throw ShareServiceError.noAppsFound(app: app, configuration: configuration)
             }
 
             try await uploadPreviews(
                 .appBundles(appPaths),
                 displayName: app,
-                version: nil,
-                bundleIdentifier: nil,
+                version: appBundles.first?.infoPlist.version.description,
+                bundleIdentifier: appBundles.first?.infoPlist.bundleId,
```

Comment:
> I find it a bit confusing that the bundle ID and version are taken from the first element in the list when uploading the previews. Is it because all the app bundles represent the same app?

Yes, that's correct. We check for that higher up: https://github.com/tuist/tuist/blob/main/Sources/TuistKit/Services/ShareService.swift#L281-L284